ApplicationCommands/src/Commands/ManagePermissions.cs

```diff
@@ -0,0 +1,141 @@
+                var perms = (await context.Client.GetApplicationCommandPermissionAsync(context.Guild.Id, command.Id))
+                    .Permissions.ToList();
+
+                var permission = perms.FirstOrDefault(cmd => cmd.Id == user.Id);
+
+                // Remove user from permission list
+                perms.Remove(permission);
+                
+                // Overwrite permissions
+                await context.Client.OverwriteGuildApplicationCommandPermissionsAsync(context.Guild.Id, command.Id, perms);
+                
+                await context.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, discordInteractionResponseBuilder.WithContent("Permission revoked!"));
+            }
+
+            [SlashCommand("del_role", "Revoke permission from a role.")]
+            public static async Task UserDel(InteractionContext context, [Option("role", "Role to revoke permission")] DiscordRole role,
+                [Option("command", "Command name")] CommandNames commandName)
+            {
+                DiscordInteractionResponseBuilder discordInteractionResponseBuilder = new() {IsEphemeral = true};
+
+                var command = GetCommand(context, commandName.ToString());
+
+                // Get all current command permissions
+                var perms = (await context.Client.GetApplicationCommandPermissionAsync(context.Guild.Id, command.Id))
+                    .Permissions.ToList();
+
+                var permission = perms.FirstOrDefault(cmd => cmd.Id == role.Id);
+
+                // Remove role from permission list
+                perms.Remove(permission);
+                
+                // Overwrite permissions
+                await context.Client.OverwriteGuildApplicationCommandPermissionsAsync(context.Guild.Id, command.Id, perms);
+                
+                await context.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, discordInteractionResponseBuilder.WithContent("Permission revoked!"));
+            }
+
+            private static DiscordApplicationCommand GetCommand(InteractionContext context, string commandName)
+            {
+                // Find the command in global and guild commands
+                KeyValuePair<ulong?, IReadOnlyList<DiscordApplicationCommand>> appCommands = context.Client.GetApplicationCommands()
+                    .RegisteredCommands.FirstOrDefault(cmd =>
+                        (cmd.Key == context.Guild.Id || cmd.Key == null) && cmd.Value.Any(x => x.Name == commandName));
+                DiscordApplicationCommand command = appCommands.Value.First(cmd => cmd.Name == commandName);
+
+                // Find the command only in guild commands
+                // DiscordApplicationCommand command = (await context.Guild.GetApplicationCommandsAsync()).FirstOrDefault(cmd => cmd.Name == commandName);
+                return command;
+            }
+        }
+
+        public enum CommandNames
```

Comment:
We can dynamically generate lists, as shown in the other examples. Shouldn't we do that here for the sake of consistency?

Reply:
Yes